ltm.go

```diff
@@ -1644,11 +1648,16 @@ func (b *BigIP) PoolMembers(name string) (*PoolMembers, error) {
 
 // AddPoolMember adds a node/member to the given pool. <member> must be in the form
 // of <node>:<port>, i.e.: "web-server1:443".
-func (b *BigIP) AddPoolMember(pool, member string) error {
+func (b *BigIP) AddPoolMember(pool, member string, partition string) error {
 	config := &poolMember{
 		Name: member,
 	}
-
+	if len(partition) > 0 {
+		// node partition
+		config.Partition = partition
```

Comment:
In this function, the node needs partition as POST argument. There is no way to do this without this.

Reply:
So actually, we have fully implemented Pool and Pool Member creation in a partition without this modification.

What you need to look at is the function getForEntity 
This calls iControlPath on the dynamic path parameter
This function takes the dynamic path and breaks apart each param, and looks if any of those individual params includes "/"  and replaces with "~"  --- what you are doing to add partition support - it's already there my friend.

func (b *BigIP) iControlPath(parts []string) string {
	var buffer bytes.Buffer
	var lastPath int
	if strings.HasPrefix(parts[len(parts)-1], "?") {
		lastPath = len(parts) - 2
	} else {
		lastPath = len(parts) - 1
	}
	for i, p := range parts {
		buffer.WriteString(strings.Replace(p, "/", "~", -1))
		if i < lastPath {
			buffer.WriteString("/")
		}
	}
	return buffer.String()
}

Reply:
So actually, we have fully implemented Pool and Pool Member creation in a partition without this modification.

What you need to look at is the function getForEntity 
This calls iControlPath on the dynamic path parameter
This function takes the dynamic path and breaks apart each param, and looks if any of those individual params includes "/"  and replaces with "~"  --- what you are doing to add partition support - it's already there my friend.

func (b *BigIP) iControlPath(parts []string) string {
	var buffer bytes.Buffer
	var lastPath int
	if strings.HasPrefix(parts[len(parts)-1], "?") {
		lastPath = len(parts) - 2
	} else {
		lastPath = len(parts) - 1
	}
	for i, p := range parts {
		buffer.WriteString(strings.Replace(p, "/", "~", -1))
		if i < lastPath {
			buffer.WriteString("/")
		}
	}
	return buffer.String()
}